helix-term/src/commands/typed.rs

```diff
@@ -260,7 +261,34 @@ fn buffer_previous(
         return Ok(());
     }
 
-    goto_buffer(cx.editor, Direction::Backward);
+    goto_buffer_by_direction(cx.editor, Direction::Backward);
+
+    Ok(())
+}
+
+/// Goto a buffer by providing <i> (index) as the argument.
+///
+/// Note that the index starts from 1.
+fn goto_buffer_by_idx(
+    cx: &mut compositor::Context,
+    args: &[Cow<str>],
+    event: PromptEvent,
+) -> anyhow::Result<()> {
+    if event != PromptEvent::Validate {
+        return Ok(());
+    }
+
+    let i_str = args
+        .first()
+        .ok_or_else(|| anyhow!("requires an argument for <i>"))?
+        .as_ref();
+    let i = i_str
+        .parse::<usize>()
+        .map_err(|_| anyhow!("'{}' is not a valid index", i_str))?;
+
+    // Convert to zero based index
+    goto_buffer_by_idx_impl(cx.editor, i - 1);
```

Comment:
This won't give feedback when the passed number is not a buffer ID. `goto_buffer_by_idx_impl` could return a result that says that the buffer ID does not exist